R/wtr-layer-segments.R

```diff
@@ -0,0 +1,134 @@
+    ## NSE to account for data argument
+    if (missing(data)) {
+      if (length(depth) <= 30) {
+        warning("Profile does not have enough readings for sm algorithm (<30): returning NA")
+        return(data.frame(min_depth = NA, nseg = NA, mld = NA, cline = NA))
+      }
+
+      if (nseg == "unconstrained") {
+        sam_list <- by_s_m(thres = thres, z0 = z0, zmax = zmax, z = depth, sigma = measure)
+        nseg <- sam_list[["nimax"]] + 1
+      } else {
+        sam_list <- by_s_m3(nr = nseg - 1, z0 = z0, zmax = zmax, z = depth, sigma = measure)
+        nseg <- nseg
+      }
+    } else {
+      if (length(data[["depth"]]) <= 30) {
+        warning("Profile does not have enough readings for sm algorithm (<30): returning NA")
+        return(data.frame(min_depth = NA, nseg = NA, mld = NA, cline = NA))
+      }
+
+      if (nseg == "unconstrained") {
+        sam_list <- eval(substitute(by_s_m(thres = thres, z0 = z0, zmax = zmax, z = depth, sigma = measure)), data)
+        nseg <- sam_list[["nimax"]] + 1
+      } else {
+        sam_list <- eval(substitute(by_s_m3(nr = nseg - 1, z0 = z0, zmax = zmax, z = depth, sigma = measure)), data)
+        nseg <- nseg
+      }
+    }
+    layers <- data.frame(
+      min_depth = z0,
+      nseg = nseg,
+      mld = sam_list[["by_s_m"]],
+      cline = cline_calc(z_seg = sam_list[["smz"]], sigma_seg = sam_list[["sms"]])
+    )
+    layers[["segments"]] <- list(data.frame(
+      segment_depth = sam_list[["smz"]],
+      segment_measure = sam_list[["sms"]]
+    ))
+    return(layers)
+  }
+
+
+#' @export
+#' @title Data filter to remove soak, heave and upcast
+#' @param z0 depth vector
+#' @param run_length Length of run upon which to start the soak removal
+#' @param index Logical: Should the function return an index value or actual value?
+#' @return index values of z0 of filtered data. Will return a warning if the function removed more than 10% of the data
+#' @description
+#' \itemize{
+#'  \item Soak period: water profiling instruments typically require a soak period where you let the instrument rest submerged at the surface. While it is "soaking" it is collecting data. We don't want that data
+#'  \item Upcast versus downcast: typically instruments are turned on before you put them in the water and turn them off once you pull them out. The data consequence of that is that you collect both the "downcast" and the "upcast". In some case the upcast is of interest but usually it isn't. And because we would prefer increasing depth data it is better to remove an upcast if it is present.
+#'  \item Heave: when lowering the instrument in rough weather a boat will heave side to side. Sometimes it will heave enough that you get small data groupings where the decreases a little while the boat heaves then go down. The overall trend is still down but those slight upticks in depth cause problems for our algorithm.
+#' }
+#'
+#' @keywords internal
```

Comment:
Is this really internal? It seems to be meant for external consumption, right?

Reply:
No this is a good catch. Fixed here: https://github.com/GLEON/rLakeAnalyzer/pull/85/commits/7e8d5fcb6e40fe30b8f0f6510d6799ecb5a71901